### PR TITLE
Pin basic-checks image, and add workflow for building it

### PR DESCRIPTION
**Pull request by tuminoid**
Pin basic-checks golang base image to have a changelog of the builds and changes, along with workflow to build it, so we get rid of locally built and pushed images.

Fixes: #864

### CODE REVIEW

**Comment by mquhuy** [comment]
LGTM

File: prow/container-images/basic-checks/README.md [hunk 1]
```diff
@@ -3,9 +3,9 @@
...
 which is used to run the basic tests in prow.
 
-## Steps to build `basic-checks` image
+## Updating image
 
-- Determine the go version to use, for e.g. `1.22`
-- Login to quay.io `docker login quay.io
-- Build the image with `docker build --build-arg GO_VERSION=1.22 -t quay.io/metal3-io/basic-checks:golang-1.22`
-- Push the image to quay: `docker push quay.io/metal3-io/basic-checks:golang-1.22`
+Make a PR and upon merging, a workflow will build and push the Quay to be used
```
Comment (kashifest) [inline prow/container-images/basic-checks/README.md]:
```suggestion
Make a PR and upon merging, a workflow will build and push the image to Quay to be used
```

**Review by kashifest — COMMENTED** [review]
/approve
one small nit inline 
/hold

File: prow/container-images/basic-checks/README.md (hunk 1 above)
Comment (tuminoid) [reply prow/container-images/basic-checks/README.md]: fixed

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Pull request merged**